src/test/java/org/jabref/logic/importer/fileformat/ImporterTestEngine.java

```diff
@@ -47,9 +49,13 @@ public static void testIsNotRecognizedFormat(Importer importer, String fileName)
         Assertions.assertFalse(importer.isRecognizedFormat(getPath(fileName), StandardCharsets.UTF_8));
     }
 
-    public static void testImportEntries(Importer importer, String fileName, String fileType) throws IOException {
-        List<BibEntry> entries = importer.importDatabase(getPath(fileName), StandardCharsets.UTF_8).getDatabase()
-                .getEntries();
+    public static void testImportEntries(Importer importer, String fileName, String fileType) throws IOException, ImportException {
+        ParserResult parserResult = importer.importDatabase(getPath(fileName), StandardCharsets.UTF_8);
+        if (parserResult.isInvalid()) {
+            throw new ImportException(parserResult.getErrorMessage());
```

Comment:
Why not throw an IOException instead? Or maybe put the call to `importDatabase` into a try-catch that converts the IOException into an ImportException? 

Overall, it would be better to throw just one Exception type. Then you don't have to change all tests to `throws Exception`, which imho is ugly, because it is so generic.

Reply:
Isn't the generic `Exception` the preferred choice for tests anyway? It has the advantage that you not need to change your tests if you decide to change the exception signature of the method under test.

I prefer to keep it that way. The error is not coming directly from the file system and thus it is not an IOException in my opinion.

Reply:
In tests it doesn't matter which exceptions are thrown